docs/source/getting-started/app-architecture.rst

```diff
@@ -82,43 +82,66 @@ The UI
 React helps us write modular UI components using a functional style - a component is rerendered whenever one of its inputs changes - with careful use of global state.
 
-The latter is especially interesting as it's how we handle ledger state in our application.
-We use a state management feature of React called `Hooks <https://reactjs.org/docs/hooks-intro.html>`_.
-We use custom DAML React hooks to query the ledger for contracts, create new contracts, and exercise choices.
+Let's see an example of a React component.
+All components are in the ``ui/src/components`` folder.
+You can navigate there within Visual Studio Code using the file *Explorer* on the left sidebar.
+We'll first look at ``App.tsx``, which is the entry point to our application.
+
+.. literalinclude:: code/ui-before/App.tsx
+  :start-after: // APP_BEGIN
+  :end-before: // APP_END
+
+An important tool in the design of our components is a React feature called `Hooks <https://reactjs.org/docs/hooks-intro.html>`_.
+Hooks allow you to share and update state across components, avoiding having to thread it through manually.
+We take advantage of hooks in particular to share ledger state across components.
+We use custom *DAML React hooks* to query the ledger for contracts, create new contracts, and exercise choices.
 
 .. TODO Link to DAML react hooks API
 
-We can see examples of this in the ``MainView`` component.
-This is the React component that enables the main functionality of the app.
+The ``useState`` hook (not specific to DAML) here keeps track of the user's credentials.
+If they are not set, we render the ``LoginScreen`` with a callback to ``setCredentials``.
+If they are set, then we render the ``MainScreen`` of the app.
+This is wrapped in the ``DamlLedger`` component, a React `context <https://reactjs.org/docs/context.html>`_ with the current view of the ledger.
+
+Let's move on to more advanced uses of our DAML React library.
+The ``MainScreen`` is a simple frame around the ``MainView`` component, which houses the main functionality of our app.
+It uses DAML React hooks to query and update ledger state.
 
 .. literalinclude:: code/ui-before/MainView.tsx
   :language: typescript
   :start-after: // USERS_BEGIN
   :end-before: // USERS_END
 
-For instance, ``allUsers`` uses the query hook to get the ``User`` contracts on the ledger.
-Note however that the query preserves privacy: only users that have added the party currently logged in are shown.
-This is because the observers of a ``User`` contract are exactly the user's friends.
+The ``useParty`` hook simply returns the current user as stored in the ``DamlLedger`` context.
+A more interesting example is the ``allUsers`` line.
+This uses the ``useStreamQuery`` hook to get all ``User`` contracts on the ledger.
+(``User`` is a TypeScript object generated from the DAML code by ``daml codegen``.)
+Note however that this query preserves privacy: only users that have added the current user have their contracts revealed.
```

Comment:
“their contracts” is a bit unclear. Maybe something like “only the `User` contracts of users that have added the current user will be returned”.